runtime/near-vm-logic/src/tests/miscs.rs

```diff
@@ -12,78 +12,74 @@ use std::{fmt::Display, fs};
 fn test_valid_utf8() {
     let mut logic_builder = VMLogicBuilder::default();
     let mut logic = logic_builder.build(get_context(vec![], false));
-    let string_bytes = "j ñ r'ø qò$`5 y'5 øò{%÷ `Võ%".as_bytes().to_vec();
-    let len = string_bytes.len() as u64;
-    logic.log_utf8(len, string_bytes.as_ptr() as _).expect("Valid utf-8 string_bytes");
+    let string = "j ñ r'ø qò$`5 y'5 øò{%÷ `Võ%";
+    let bytes = logic.internal_mem_write(string.as_bytes());
+    logic.log_utf8(bytes.len, bytes.ptr).expect("Valid UTF-8 in bytes");
     let outcome = logic.compute_outcome_and_distribute_gas();
-    assert_eq!(outcome.logs[0], String::from_utf8(string_bytes).unwrap());
+    assert_eq!(outcome.logs[0], string);
     assert_costs(map! {
         ExtCosts::base: 1,
         ExtCosts::log_base:  1,
-        ExtCosts::log_byte: len,
+        ExtCosts::log_byte: bytes.len,
         ExtCosts::read_memory_base: 1,
-        ExtCosts::read_memory_byte: len,
+        ExtCosts::read_memory_byte: bytes.len,
         ExtCosts::utf8_decoding_base: 1,
-        ExtCosts::utf8_decoding_byte: len,
+        ExtCosts::utf8_decoding_byte: bytes.len,
     });
 }
 
 #[test]
 fn test_invalid_utf8() {
     let mut logic_builder = VMLogicBuilder::default();
     let mut logic = logic_builder.build(get_context(vec![], false));
-    let string_bytes = [128].to_vec();
-    let len = string_bytes.len() as u64;
-    assert_eq!(logic.log_utf8(len, string_bytes.as_ptr() as _), Err(HostError::BadUTF8.into()));
+    logic.internal_mem_write(b"\x80");
+    assert_eq!(logic.log_utf8(1, 0), Err(HostError::BadUTF8.into()));
```

Comment:
Similar to what you did above, this should use the return values of `internal_mem_write` instead of hardcoded 1,0, even though we can know it’s being placed at the beginning of the buffer it feels wrong to me